Mathlib/CategoryTheory/ChosenFiniteProducts.lean

```diff
@@ -563,6 +568,26 @@ end prodComparison
 
 end ChosenFiniteProductsComparison
 
+open Limits
+
+variable {P : C → Prop}
+
+-- TODO: Introduce `ClosedUnderFiniteProducts`?
+/-- The restriction of a cartesian-closed category along an object property that's closed under
+finite products is cartesian-closed. -/
+noncomputable def fullSubcategory (hP₀ : ClosedUnderLimitsOfShape (Discrete PEmpty) P)
+    (hP₂ : ClosedUnderLimitsOfShape (Discrete WalkingPair) P) :
+    ChosenFiniteProducts (FullSubcategory P) where
+  product X Y := {
+    cone := BinaryFan.mk
+      (P := ⟨X.1 ⊗ Y.1, hP₂ (product X.obj Y.obj).isLimit <| by rintro ⟨_ | _⟩ <;> simp [X.2, Y.2]⟩)
+      (fst X.1 Y.1) (snd X.1 Y.1)
+    isLimit := isLimitOfReflectsOfMapIsLimit (fullSubcategoryInclusion _) _ _ <|
+      (product X.obj Y.obj).isLimit.ofIsoLimit <| isoBinaryFanMk _
```

Comment:
It would be slightly better to obtain this `isLimit` more directly so that lemmas like `lift_def` become `by rfl`.

Reply:
Do you have a suggestion how? Maybe this has to wait on Bhavik's comment: https://github.com/leanprover-community/mathlib4/pull/23562#discussion_r2023730782

Reply:
Perhaps just `BinaryFan.IsLimit.mk`?

Reply:
Wow, it's so much better now! Everything is rfl!